### DISCUSSION TITLE
Can I make a new programming language based on csharplang

### DISCUSSION THREAD

**Discussion by BoQsc** [dotnet/csharplang#4868] [Q&A]
The C# specification clearly states that C# language specification is copyrighted.
https://github.com/dotnet/csharplang/blob/main/spec/LICENSE.md

## The questions are: 
Why csharplang is copyrighted and not in a public domain or under open source license 
or public copyright license like Creative Commons.

Why the implemented compiler https://github.com/dotnet/roslyn 
is under MIT license but not the csharplang specification?

**Comment by YairHalberstadt** [comment]
The specification document is under copyright, but the specification as an abstract concept is not. In other words you are free to modify csharp however you like, but you may legally have to write the specification document from scratch yourself, or use the ecma c#5 specification as a starting point.
